src/Shipment.ts

```diff
@@ -0,0 +1,73 @@
+import { AirEastShipper, ChicagoSprintShipper, PacificParcelShipper, Shipper } from "./Shipper";
+import { Companies, Order } from "./types";
+import { zipCodesMap } from "./zipCodesMap";
+
+
+export interface Shipment {
+  getInstance?: (order: Order) => Shipment;
+  getShipmentID?: () => number;
+  ship: () => string;
+}
+export class BaseShipment implements Shipment {
+  private static int = 1;
+  private static shipment: Shipment;
+  private shipper: Shipper;
+
+  private weight: number;
+  private fromAddress: string;
+  private fromZipCode: string;
+  private toAddress: string;
+  private toZipCode: string;
+  private shipmentID: number;
+
+
+  constructor({ weight, fromAddress, fromZipCode, toAddress, toZipCode, shipmentID }: Order) {
+    this.weight = weight;
+    this.fromAddress = fromAddress;
+    this.fromZipCode = fromZipCode;
+    this.toAddress = toAddress;
+    this.toZipCode = toZipCode;
+    this.shipmentID = shipmentID === 0 ? this.getShipmentID() : shipmentID;
+    this.getShipper();
+  }
+
+  private getShipper() {
+    if (!this.fromZipCode) {
```

Comment:
Shipper is singleton as well, so you need to call here Shipper.getInstance() for each kind of shipper. 
You don't store anything in the Shipper class itself, it means we can reuse this class (class family), no need to create new one each time the new order has to be delivered